File: Classification/preprocessing/logistic_regresssion_data.r
df = read.csv("../train_ISIC_2020.csv", header = TRUE)

## general
logitMod <- glm(target ~ sex + age_approx + anatom_site_general_challenge, data=df, family=binomial(link="logit"))
summary(logitMod)

## age
logitMod2 <- glm(target ~ age_approx, data=df, family=binomial(link="logit"))
summary(logitMod2)

## sex (no effect)
logitMod3 <- glm(target ~ sex, data=df, family=binomial(link="logit"))
summary(logitMod3)

## age+sex
logitMod4 <- glm(target ~ sex+age_approx, data=df, family=binomial(link="logit"))
summary(logitMod4)
##  ln(p/(1-p)) = -6.607236 + 0.307200*(if male) + 0.045152*age 
## ⇒ p = σ(+6.607236 - 0.307200*(if male) - 0.045152*age)
## ⇒ p = 1/(1+exp(+6.607236 - 0.307200*(if male) - 0.045152*age))




 

File: Classification/preprocessing/logistic_regresssion_data.py
import numpy as np
import pandas as pd
import sklearn
import sklearn.linear_model
from sklearn.impute import SimpleImputer

# the data
df = pd.read_csv('../train_ISIC_2020.csv')
X = pd.concat([pd.get_dummies(df['sex']),
               pd.get_dummies(df['anatom_site_general_challenge']).iloc[:,0:5],
               df['age_approx']],axis=1,ignore_index=True)
X = pd.concat([pd.get_dummies(df['sex']),
               pd.get_dummies(df['anatom_site_general_challenge']),
               df['age_approx']],axis=1)
y = df['target']
# remove missing data with the mean
imp = SimpleImputer(missing_values=np.nan, strategy='mean')
imp.fit(X)
X_bis = imp.transform(X)
# fitting
theModel = sklearn.linear_model.LogisticRegressionCV()
theModel.fit(X_bis,y)
# print theModel.coef_


## tp = np.array([np.ones(33126),X_bis[:,8]])
## theModel.fit(tp.T,y)


import statsmodels.api as sm
model2 = sm.Logit(y,X_bis)
result = model2.fit(method='newton')
